meinberlin/apps/projects/tasks.py

```diff
@@ -0,0 +1,169 @@
+        list_next_projects_start = get_next_projects_start()
+
+    msg = "Future projects becoming active in next 10mins will be removed from cache: "
+
+    if list_next_projects_start:  # if not an empty list
+        # set the redis key: value
+        cache.set("next_projects_start", list_next_projects_start)
+
+        for str_project_starts in list_next_projects_start:
+            # convert to datetime obj
+            date_project_starts = parser.parse(str_project_starts)
+
+            # compare now with next start date
+            now = datetime.now(tz=timezone.utc)
+            remain_time = date_project_starts - now
+
+            if remain_time <= timedelta(minutes=10):
+                # schedule cache clear for the remaining time btwn now and next start
+                reset_cache_for_projects.apply_async(
+                    [True, False], countdown=remain_time
+                )
+                msg += f"""
+                    {date_project_starts} in {remain_time} minutes
+                    """
+    else:
+        msg += "None"
+    return logger.info(msg)
+
+
+@shared_task(name="schedule_reset_cache_for_projects_becoming_past")
+def schedule_reset_cache_for_projects_becoming_past():
+    """The task is set via celery beat every 10 minutes in
+    settings/production.txt.
+
+    Returns:
+        A string message if the next project ends
+        date will be removed from the cache.
+    """
+
+    # check if redis has a next_project_ends key
+    str_project_ends = cache.get("next_project_ends")
+
+    if not str_project_ends:
+        str_project_ends = get_next_project_ends()
+        if not str_project_ends:
+            raise Exception("At least one phase with dates is required")
+        # set the redis key: value
+        cache.set("next_project_ends", str_project_ends)
+
+    msg = "Active projects will be removed from cache: "
+
+    # compare now with next end date
+    date_project_ends = parser.parse(str_project_ends)
+    now = datetime.now(tz=timezone.utc)
+    remain_time = now - date_project_ends
+    if remain_time <= timedelta(minutes=10):
+        # schedule cache clear for the remaining time btwn now and next end
+        reset_cache_for_projects.apply_async([False, True], countdown=remain_time)
+        msg = f"""
+                {date_project_ends} in {remain_time} minutes
+                """
+    else:
+        msg += "None"
+    return logger.info(msg)
+
+
+@shared_task
+def reset_cache_for_projects(starts, ends):
+    msg = "Clear cache "
+    if starts:
+        # remove redis key next_project_start
+        cache.delete("next_project_starts")
+        cache.delete_many(
+            [
+                "projects_activeParticipation",
+                "projects_futureParticipation",
+                "private_projects",
+                "extprojects",
+            ]
+        )
+        if cache.get("projects_*") or cache.get("extprojects"):
```

Comment:
`cache.get("projects_*")` will only work with redis, not with any of the other caching backends, right?

Reply:
hmm, I can't find any information on this though. The django docs don't mention wildcards (or I'm too stupid to find them), only the django-redis package mentions it. However if we want to be on the safe side I think we can use `cache.get_many(['projects_activeParticipation', 'projects_pastParticipation', ... , 'extprojects',])`